File: snippets/fsharp/VS_Snippets_CLR_System/System.Convert.ToBase64String/fs/ToBase64String3.fs
// <Snippet3>
open System

// Define a byte array.
let bytes = 
    [| for i = 0 to 99 do byte (i + 1) |]

let originalTotal = Array.sumBy int bytes

// Display summary information about the array.
printfn "The original byte array:"
printfn $"   Total elements: {bytes.Length}"
printfn $"   Length of String Representation: {BitConverter.ToString(bytes).Length}"
printfn $"   Sum of elements: {originalTotal:N0}"
printfn ""

// Convert the array to a base 64 string.
let s = Convert.ToBase64String(bytes, Base64FormattingOptions.InsertLineBreaks)
printfn $"The base 64 string:\n   {s}\n"

// Restore the byte array.
let newBytes = Convert.FromBase64String s

let newTotal = Array.sumBy int newBytes

// Display summary information about the restored array.
printfn $"   Total elements: {newBytes.Length}"
printfn $"   Length of String Representation: {BitConverter.ToString(newBytes).Length}"
printfn $"   Sum of elements: {newTotal:N0}"

// The example displays the following output:
//   The original byte array:
//      Total elements: 100
//      Length of String Representation: 299
//      Sum of elements: 5,050
//
//   The base 64 string:
//      AQIDBAUGBwgJCgsMDQ4PEBESExQVFhcYGRobHB0eHyAhIiMkJSYnKCkqKywtLi8wMTIzNDU2Nzg5
//   Ojs8PT4/QEFCQ0RFRkdISUpLTE1OT1BRUlNUVVZXWFlaW1xdXl9gYWJjZA==
//
//      Total elements: 100
//      Length of String Representation: 299
//      Sum of elements: 5,050
// </Snippet3>

File: snippets/csharp/System/Base64FormattingOptions/Overview/ToBase64String3.cs
// <Snippet3>
using System;

public class Base64RoundTripExample
{
    public static void Main()
    {
        // Define a byte array.
        byte[] bytes = new byte[100];
        int originalTotal = 0;
        for (int ctr = 0; ctr <= bytes.GetUpperBound(0); ctr++)
        {
            bytes[ctr] = (byte)(ctr + 1);
            originalTotal += bytes[ctr];
        }
        // Display summary information about the array.
        Console.WriteLine("The original byte array:");
        Console.WriteLine($"   Total elements: {bytes.Length}");
        Console.WriteLine($"   Length of String Representation: {BitConverter.ToString(bytes).Length}");
        Console.WriteLine($"   Sum of elements: {originalTotal:N0}");
        Console.WriteLine();

        // Convert the array to a base 64 string.
        string s = Convert.ToBase64String(bytes,
                                          Base64FormattingOptions.InsertLineBreaks);
        Console.WriteLine($"The base 64 string:\n   {s}\n");

        // Restore the byte array.
        byte[] newBytes = Convert.FromBase64String(s);
        int newTotal = 0;
        foreach (byte newByte in newBytes)
        {
            newTotal += newByte;
        }

        // Display summary information about the restored array.
        Console.WriteLine($"   Total elements: {newBytes.Length}");
        Console.WriteLine($"   Length of String Representation: {BitConverter.ToString(newBytes).Length}");
        Console.WriteLine($"   Sum of elements: {newTotal:N0}");
    }
}
// The example displays the following output:
//   The original byte array:
//      Total elements: 100
//      Length of String Representation: 299
//      Sum of elements: 5,050
//
//   The base 64 string:
//      AQIDBAUGBwgJCgsMDQ4PEBESExQVFhcYGRobHB0eHyAhIiMkJSYnKCkqKywtLi8wMTIzNDU2Nzg5
//   Ojs8PT4/QEFCQ0RFRkdISUpLTE1OT1BRUlNUVVZXWFlaW1xdXl9gYWJjZA==
//
//      Total elements: 100
//      Length of String Representation: 299
//      Sum of elements: 5,050
// </Snippet3>
